queue/src/main/java/org/killbill/bus/DefaultPersistentBus.java

```diff
@@ -141,35 +141,55 @@ public DefaultPersistentBus(final DataSource dataSource, final Properties proper
              new MetricRegistry(), new DatabaseTransactionNotificationApi());
     }
 
-    @Override
-    public void start() {
 
+    @Override
+    public boolean initQueue() {
         if (config.isProcessingOff()) {
-            log.warn("PersistentBus processing is off, does not start");
-            return;
+            log.warn("PersistentBus processing is off, cannot be initialized");
+            return false;
         }
 
         if (isInitialized.compareAndSet(false, true)) {
             dao.initialize();
             dispatcher.start();
-            startQueue();
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    @Override
+    public boolean startQueue() {
 
+        if (config.isProcessingOff()) {
+            log.warn("PersistentBus processing is off, cannot be started");
+            return false;
+        }
+
+        if (!isInitialized.get()) {
+            // Make it easy for our tests, so they simply call startQueue
+            initQueue();
+        }
+
+        if (isStarted.compareAndSet(false, true)) {
             if (config.getPersistentQueueMode() == PersistentQueueMode.STICKY_POLLING || config.getPersistentQueueMode() == PersistentQueueMode.STICKY_EVENTS) {
                 reaper.start();
             }
-            isStarted.set(true);
+            super.startQueue();
+            return true;
+        } else {
+            return false;
         }
-
     }
 
     @Override
-    public void stop() {
+    public void stopQueue() {
         if (isStarted.compareAndSet(true, false)) {
             isInitialized.set(false);
+            super.stopQueue();
+            reaper.stop();
```

Comment:
Not sure if it matters but shouldn't the reaper be stopped first (so it doesn't move entries after the queue is stopped)? Also, shouldn't the `dao` be the last one to stop?